### ISSUE TITLE
typescript 2.3.0 for-await-of should not be red

### ISSUE THREAD

**Issue by xmlking** [palantir/tslint#2633]
typescript 2.3.0 now support async generators and  for-await-of
https://blogs.msdn.microsoft.com/typescript/2017/04/10/announcing-typescript-2-3-rc/
TSLint show not consider it error/red
![image](https://cloud.githubusercontent.com/assets/627951/25361995/2d0f5ca4-2906-11e7-9b73-e457e59b2fee.png)

**Comment by ajafff** [comment]
The issue template you removed while posting this issue contains useful hints what information is required for us to analyse and possibly resolve your issue.
And please provide the error message you get from tslint and the name of the rule.

``````
### Bug Report

- __TSLint version__:
- __TypeScript version__:
- __Running TSLint via__: (pick one) CLI / Node.js API / VSCode / grunt-tslint / Atom / Visual Studio / etc

#### TypeScript code being linted

```ts
// code snippet
```

with `tslint.json` configuration:

```json

```

#### Actual behavior

#### Expected behavior
``````
